### PR TITLE
Consolidate benchmarks

### PR DESCRIPTION
**Pull request by andygrove**
Follows on from https://github.com/apache/arrow-datafusion/pull/23.

There is now a single benchmark crate that can run against DataFusion or Ballista.

Integration tests pass.

Closes https://github.com/apache/arrow-datafusion/issues/6.

### CODE REVIEW

**Comment by andygrove** [comment]
@alamb @jorgecarleitao @Dandandan This is rebased and ready for review

File: benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -181,832 +202,97 @@ async fn benchmark(opt: BenchmarkOpt) -> Result<Vec<arrow::record_batch::RecordB
...
-                                c_acctbal > 0.00
-                                and substring(c_phone from 1 for 2) in
-                                    ('13', '31', '23', '29', '30', '18', '17')
-                        )
-                        and not exists (
-                            select
-                                *
-                            from
-                                orders
-                            where
```
Comment (Dandandan) [inline benchmarks/src/bin/tpch.rs:998]: :+1: :+1:

**Review by Dandandan — APPROVED** [review]
Great, I think this looks good

**Review by jorgecarleitao — APPROVED** [review]
Brutal ❤️ 💯

**Pull request merged**